scripts/wrap.sh

```diff
@@ -1,27 +1,79 @@
 #!/bin/bash
+
+set -e  # Exit on any error
+
+# Determine rootdir based on our script location
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+ROOTDIR="$(dirname $DIR)"
+
+# Detect luamin binary
+LUAMIN="luamin"
+if ! which $LUAMIN 1> /dev/null; then
+    LUAMIN="${ROOTDIR}/scripts/node_modules/.bin/luamin"
+fi
+if ! which $LUAMIN 1> /dev/null; then
+    echo "Need luamin in PATH (do \`npm install -g luamin\` or run \`npm install\` from the scripts/ folder)"; exit 1
+fi
+
+# Parse args, or use defaults
+MINIFY="${1:-false}"
 # We expect this file to be run from the <repo>/scripts directory
-LUA_SRC=../src/ButtonHUD.lua
-CONF_DST=../ButtonHUD.conf
-mkdir -p work
+LUA_SRC=${2:-$ROOTDIR/src/ButtonHUD.lua}
+CONF_DST=${3:-$ROOTDIR/ButtonHUD.conf}
+
+# Make a fresh work dir
+WORK_DIR=${ROOTDIR}/scripts/work
+(rm -rf $WORK_DIR || true) && mkdir -p $WORK_DIR
 
 # Extract the exports because the minifier will eat them.
-grep "\-- \?export:" $LUA_SRC | sed -e 's/^[ \t]*/        /' -e 's/-- export:/--export:/' > work/ButtonHUD.exports
+grep "\-- \?export:" $LUA_SRC | sed -e 's/^[ \t]*/        /' -e 's/-- export:/--export:/' > $WORK_DIR/ButtonHUD.exports
 VERSION_NUMBER=`grep "VERSION_NUMBER = .*" $LUA_SRC | sed -E "s/\s*VERSION_NUMBER = (.*)/\1/"`
 
-sed "/-- \?export:/d" $LUA_SRC > work/ButtonHUD.extracted.lua
+if [[ "${VERSION_NUMBER}" == "" ]]; then
+    echo "Failed to detect version number"; exit 1
```

Comment:
For errors, could you add an "ERROR:" tag, which we can then put in VSCode to detect and highlight errors?  Can add a problemMatcher to the tasks.json then.

Reply:
And for that matter, check the output code for all tools we run here - if any of them fail, we should spit an error.